plone/app/contenttypes/indexers.py

```diff
@@ -153,6 +153,16 @@ def getObjSize_file(obj):
 
 
 @indexer(IDexterityContent)
+def MimeType(obj):
+    if obj.portal_type in ['File', 'Image']:
+        try:
+            return obj.content_type()
+        except AttributeError:
```

Comment:
It is ok if an indexer raises an attribute error. the index detects it and does not catalog anything then. so do not catch it.